Tempura/UITests/ViewControllerTestCase.swift

```diff
@@ -59,86 +59,115 @@ public protocol ViewControllerTestCase {
   
   /// configure the VC for the specified `testCase`
   /// this is typically used to manually inject the ViewModel to all the children VCs.
-  func configure(vc: VC, for testCase: String)
+  func configure(vc: VC, for testCase: String, model: VC.V.VM)
 }
 
 
 public extension ViewControllerTestCase where Self: XCTestCase {
-  func uiTest(testCases: [String], context: UITests.VCContext<VC>) {
-    let snapshotConfiguration = UITests.VCScreenSnapshot<VC>(
-      vc: self.viewController,
-      container: context.container,
-      testCases: testCases,
-      hooks: context.hooks,
-      size: context.screenSize
-    )
-    
-    let viewControllers = snapshotConfiguration.renderingViewControllers
+  func uiTest(testCases: [String: VC.V.VM], context: UITests.VCContext<VC>) {
     let screenSizeDescription: String = "\(UIScreen.main.bounds.size)"
-    
-    var expectations: [XCTestExpectation] = []
-    
-    for (identifier, vcs) in viewControllers {
+    let descriptions: [String: String] = Dictionary(uniqueKeysWithValues: testCases.keys.map { identifier in
       let description = "\(identifier) \(screenSizeDescription)"
-      
-      let expectation = XCTestExpectation(description: description)
-      XCUIDevice.shared.orientation = context.orientation
-      
-      let isViewReadyClosure: (UIView) -> Bool = { view in
-        var isOrientationCorrect = true
-        
-        // read again in case some weird code changed it outside the ViewControllerTestCase APIs
-        let isViewInPortrait = view.frame.size.height > view.frame.size.width
-        
-        if context.orientation.isPortrait {
-          isOrientationCorrect = isViewInPortrait
-          
-        } else if context.orientation.isLandscape {
-          isOrientationCorrect = !isViewInPortrait
+      return (identifier, description)
+    })
+    let expectations: [String: XCTestExpectation] = descriptions.mapValues { identifier in
+      return XCTestExpectation(description: description)
+    }
+
+    XCUIDevice.shared.orientation = context.orientation
+
+    DispatchQueue.global().async {
+      for (identifier, model) in testCases {
+        var contained: VC?
+        var container: UIViewController?
+        DispatchQueue.main.sync {
+          contained = self.viewController
+          container = context.container.container(for: contained as! UIViewController)
```

Comment:
We could enforce VC to also be a `UIViewController` to avoid this force casting:
```swift
public protocol ViewControllerTestCase {
  associatedtype VC: AnyViewController & UIViewController
  ...
}
```

Then (together with the implicit unwrapping discussed in the previous comment):
```swift
container = context.container.container(for: contained)
```